web/src/components/citations.tsx

```diff
@@ -41,33 +39,29 @@ export const ProcessText: (text: string, base_count: number) => [string, Map<str
     /\[\s*([a-z]+)\s*\]/g,
     (_match: string, x: string) => `[${x}]`
   )
+  return response;
+}
 
-  // -------------- map citations from strings into numbers --------------
-
+export const findCitations: (text: string, citations: Citations[]) => Map<string, Citation> = (text, citations) => {
   // figure out what citations are in the response, and map them appropriately
-  const cite_map = new Map<string, number>();
-  let cite_count = 0;
+  const cite_map = new Map<string, Citation>();
 
   // scan a regex for [x] over the response. If x isn't in the map, add it.
   // (note: we're actually doing this twice - once on parsing, once on render.
   // if that looks like a problem, we could swap from strings to custom ropes).
   const regex = /\[([a-z]+)\]/g;
```

Comment:
(minor) this supports only letters and skips numbers .. is that on purpose?

Reply:
yes :/ There is a previous function that tries to clean up the text, but the idea is that the backend returns citations only as letters

Reply:
in that case comments in other places in this repo about "this can be easily changed to numbers" should ideally point to this line :D

but I guess someone would notice if it stopped working after that change, so not a big deal

Reply:
"Easily" is "easily if you know exactly where to find the magical 3 lines and what to change" :D